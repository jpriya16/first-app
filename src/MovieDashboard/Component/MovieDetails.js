import React from "react";
export default function MovieDetails(Movies){
    return(
        <div>
            <table>
                <tr>
                    <th>Movie name</th>
                    <th>ID</th>
                    <th>Total tickets</th>
                    <th>Booked</th>
                </tr>
                { Movies.map( (val, key) => {
                    return (
                        <tr key={key}>
                            <td>{val.name}</td>
                            <td>{val._id}</td>
                            <td>{val.total_tickets}</td>
                            <td>{val.booked_tickets}</td>
                        </tr>
                    )
                }) }
            </table>
        </div>
    )
}
