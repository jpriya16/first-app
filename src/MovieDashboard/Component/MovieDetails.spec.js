/* eslint-disable */


import React from "react";
import { shallow } from 'enzyme';
import MovieDetails from "./MovieDetails";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const movies =
    [
        {
            "_id": "624158496f047803e8ae1eb8",
            "name": "Movie5",
            "total_tickets": 40,
            "booked_tickets": 10
        }
    ]

describe("Get Movie Details",()=>{
    it("should Match the snapshot", ()=>{
        const component = shallow(MovieDetails(movies))
        expect(component.getElements()).toMatchSnapshot();
    })
})
