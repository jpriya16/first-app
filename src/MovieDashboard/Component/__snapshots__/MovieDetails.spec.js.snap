// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Movie Details should Match the snapshot 1`] = `
Array [
  <div>
    <table>
      <tr>
        <th>
          Movie name
        </th>
        <th>
          ID
        </th>
        <th>
          Total tickets
        </th>
        <th>
          Booked
        </th>
      </tr>
      <tr>
        <td>
          Movie5
        </td>
        <td>
          624158496f047803e8ae1eb8
        </td>
        <td>
          40
        </td>
        <td>
          10
        </td>
      </tr>
    </table>
  </div>,
]
`;
